File: app/src/main/java/com/octo/androidweather/CityForecastController.java
package com.octo.androidweather;

import com.octo.usecases.CityForecastInteractor;

class CityForecastController {
    private final CityForecastInteractor interactor;

    public CityForecastController(final CityForecastInteractor interactor) {
        this.interactor = interactor;
    }

    public void loadCityForecast(String city) {
        interactor.loadCityForecast(trimNonNullString(city));
    }

    private static String trimNonNullString(String string) {
        return string != null ? string.trim() : null;
    }
}

File: app/src/main/java/com/octo/androidweather/CityForecastController.kt
package com.octo.androidweather

import com.octo.usecases.CityForecastInteractor

class CityForecastController(private val interactor: CityForecastInteractor) {
    fun loadCityForecast(city: String) {
        interactor.loadCityForecast(city)
    }
}